### ISSUE TITLE
Processing pixel data can be quite slow.

### ISSUE THREAD

**Issue by jakecjones** [The-Lunar-Curve/SBC-POC#2]
I think we could probably consolidate a few of these methods by clustering the data as it's loaded initially. We tend to cycle through these larger arrays a few times before choosing the dominant colors. Maybe we can divide this data better before the clustering step?

**Comment by elishacoad** [comment]
Yes, could also be useful if we decide to add other features like pixalizing the image or wanting to find the average of an area of the image.
